File: src/main/java/checkout/Checkout.java
package checkout;

class Checkout {
    private int total;
    private int numberOfA = 0;
    private int numberOfB = 0;
    private Receipt receipt = new Receipt();

    void scan(String sku) {
        if ("A".equals(sku)) {
            total += 50;
            receipt.scannedA();
        } else if ("B".equals(sku)) {
            total += 30;
            receipt.scannedB();
        } else if ("C".equals(sku)) {
            total += 20;
            receipt.scannedC();
        } else if ("D".equals(sku)) {
            total += 15;
            receipt.scannedD();
        }
        if ("A".equals(sku)) {
            numberOfA++;
            if (numberOfA % 3 == 0) {
                total -= 20;
            }
        } else if ("B".equals(sku)) {
            numberOfB++;
            if (numberOfB % 2 == 0) {
                total -= 15;
            }
        }
    }

    int total() {
        return total;
    }

    public String receipt() {
        return receipt.text();
    }
}

File: exercises/dotnet/McrDigital.Bootcamp1.Checkout/Checkout.cs
namespace McrDigital.Bootcamp1.Checkout
{
    public class Checkout
    {
        private int _total;
        private int _numberOfA;
        private int _numberOfB;
        private Receipt _receipt = new Receipt();

        public string Receipt
        {
            get => _receipt.Text;
        }

        public int Total
        {
            get => _total;
        }

        public void Scan(string sku)
        {
            if ("A".Equals(sku))
            {
                _total += 50;
                _receipt.ScannedA();
            }
            else if ("B".Equals(sku))
            {
                _total += 30;
                _receipt.ScannedB();
            }
            else if ("C".Equals(sku))
            {
                _total += 20;
                _receipt.ScannedC();
            }
            else if ("D".Equals(sku))
            {
                _total += 15;
                _receipt.ScannedD();
            }

            if ("A".Equals(sku))
            {
                _numberOfA++;
                if (_numberOfA % 3 == 0)
                {
                    _total -= 20;
                }
            }
            else if ("B".Equals(sku))
            {
                _numberOfB++;
                if (_numberOfB % 2 == 0)
                {
                    _total -= 15;
                }
            }
        }
    }
}